docs: shorten README and remove other docs (#159)
Redocly/openapi-cli: README.md

Currently, @redocly/openapi-cli supports these features:

- [x] Multifile validation. No need to bundle your file before validation.
- [x] Lightning-fast validation. Lint a 1 Mb file in less than one second.
- [x] Built-in rules for common validations.
- [x] Configurable severity levels for each rule.
- [x] Human-readable error messages with codeframes and stylish format options.
- [x] Intuitive suggestions for misspelled types or references.

> - [x] Easy to implement custom rules. 

Reviewer: link to custom rules docs?

Author: There is a link to the docs below. When we have solid custom rules docs, we can add a deep link here to those.

Reviewer: Good

- [x] Bundle a multifile definition into a single file.
- [x] Decorators to modify a validated definition during bundling.
- [x] Preview reference docs for local development.
- [x] Support for OpenAPI 2 (fka Swagger) and OpenAPI 3.0.
- [ ] Support for OpenAPI 3.1 ([coming soon](https://github.com/Redocly/openapi-cli/issues/160)).

## What makes this tool different

> Unlike other OpenAPI linters, `@redocly/openapi-cli` defines the possible type tree of a valid OpenAPI definition and then traverses it. This approach is very similar to how compilers work and results in major performance benefits over other approaches. Extend functionality at different points in the lifecycle with preprocessors, rules, and decorators by following the [visitor pattern](https://en.wikipedia.org/wiki/Visitor_pattern).

Reviewer: ```suggestion
Unlike other OpenAPI linters, `@redocly/openapi-cli` defines the possible type tree of a valid OpenAPI definition and then traverses it. This approach is very similar to how linters for programming languages work and results in major performance benefits over other approaches. Extend functionality at different points in the lifecycle with preprocessors, rules, and decorators.
```

> ## [Read the docs](https://redoc.ly/docs/cli/)

Reviewer: I would also add TLDR usage section, basically

`npx @redocly/openapi-cli lint path-to-file.yaml`
